### PR TITLE
Simplified multitenancy checking for Payara.

### PR DESCRIPTION
**Pull request by pveentjer**
Instead of pushing the complexity into the OperationRunner and OperationThread, it is now moved to the TenantAwareOperationRunner.

Check the following issue for more details:

https://github.com/hazelcast/hazelcast/issues/23559

### CODE REVIEW

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java [hunk 1]
```diff
@@ -106,13 +106,14 @@ public class OperationRunnerImpl extends OperationRunner implements StaticMetric
 
     static final int AD_HOC_PARTITION_ID = -2;
 
+    final OperationServiceImpl operationService;
+    final Counter executedOperationsCounter;
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java:110]: Can you also move the `@Probe(name = OPERATION_METRIC_OPERATION_RUNNER_EXECUTED_OPERATIONS_COUNT, level = DEBUG)`  line above this field? The annotation is now on `thisAddress` field and all tests fail.

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/OperationRunner.java [hunk 2]
```diff
@@ -58,19 +58,16 @@ public OperationRunner(int partitionId) {
      * timed out or has run successfully
      * @throws Exception if there was an exception raised while processing the packet
      */
-    public abstract boolean run(Packet packet) throws Exception;
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/OperationRunner.java:61]: Revert to old behavior.

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationThread.java [hunk 3]
```diff
@@ -132,11 +132,10 @@ public final void executeRun() {
 
     private void process(Object task) {
         try {
-            boolean putBackInQueue = false;
             if (task.getClass() == Packet.class) {
-                putBackInQueue = process((Packet) task);
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationThread.java:137]: Reverting to old behavior.

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java [hunk 4]
```diff
@@ -467,7 +460,7 @@ private void logOperationError(Operation op, Throwable e) {
     }
 
     @Override
-    public boolean run(Packet packet) throws Exception {
+    public void run(Packet packet) throws Exception {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java:463]: reverting to old behavior

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java [hunk 5]
```diff
@@ -261,37 +258,33 @@ private boolean run(Operation op, long startNanos) {
...
-                return true;
-            }
+            op.beforeRun();
+            call(op);
         } catch (Throwable e) {
             handleOperationError(op, e);
         } finally {
-            op.afterRunFinal();
-            if (publishCurrentTask) {
+             if (publishCurrentTask) {
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java:269]: we are missing here the call to `op.afterRunFinal()`

Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java]: Fixed.

**Review by vbekiaris — APPROVED** [review]

**Pull request merged**